Bento/Renderable/CustomInputComponent.swift

```diff
@@ -0,0 +1,119 @@
+        self.customInput = customInput
+        self.highlightColor = highlightColor
+        self.backgroundColor = backgroundColor
+        self.base = source
+        self.focusEligibility = .eligible(contentStatus)
+    }
+
+    func render(in view: ComponentView) {
+        view.inputNodes = customInput
+        view.backgroundColor = backgroundColor
+        view.highlightingGesture.didTap = .manual
+        view.highlightingGesture.highlightColor = highlightColor
+
+        base.render(in: view.containedView)
+    }
+}
+
+extension CustomInputComponent {
+    final class ComponentView: InteractiveView, FocusableView {
+        var inputNodes: CustomInput? {
+            didSet {
+                guard isFirstResponder else { return }
+                if let nodes = inputNodes {
+                    customInputView?.update(nodes)
+                    reloadInputViews()
+                } else {
+                    _ = resignFirstResponder()
+                }
+            }
+        }
+
+        var customInputView: InputView?
+        var focusToolbar: FocusToolbar?
+        let containedView = Base.View.generate() as! Base.View
+
+        override init(frame: CGRect) {
+            super.init(frame: frame)
+
+            containedView.add(to: self).pinEdges(to: self)
+        }
+
+        @available(*, unavailable)
+        required init?(coder aDecoder: NSCoder) { fatalError() }
+
+        public override var inputView: UIView? {
+            return customInputView
+        }
+
+        public override var inputAccessoryView: UIView? {
+            return focusToolbar
+        }
+
+        public override func becomeFirstResponder() -> Bool {
+            if let nodes = inputNodes {
+                customInputView = InputView()
+                focusToolbar = FocusToolbar(view: self)
+                customInputView!.update(nodes)
+            }
+
+            if super.becomeFirstResponder() {
+                highlightingGesture.isHighlighted = true
+
+                NotificationCenter.default.addObserver(
+                    self,
+                    selector: #selector(keyboardDidDisappear),
+                    name: UIResponder.keyboardDidHideNotification,
+                    object: nil
+                )
+
+                return true
+            }
+
+            customInputView = nil
+            focusToolbar = nil
+            return false
+        }
+
+        public override func resignFirstResponder() -> Bool {
+            highlightingGesture.isHighlighted = false
+            return super.resignFirstResponder()
+        }
+
+        public func focus() {
+            _ = becomeFirstResponder()
+        }
+
+        private func neighboringFocusEligibilityDidChange() {
+            focusToolbar?.updateFocusEligibility(with: self)
+            reloadInputViews()
+        }
+
+        @objc func keyboardDidDisappear() {
+            if isFirstResponder { _ = resignFirstResponder() }
+            customInputView = nil
```

Comment:
Why do we need to make it `nil`?

Reply:
Release unused view.

Reply:
This is copy pasta from DetailedDescription anyway. :)

Reply:
I'm also not sure why do we need to observe key board notifications

Reply:
According to my vague memory, these need to stay until the system input view proxy thingy is done with the view.

Reply:
Argh. The toolbar would stick around after interactive swipe-to-dismiss-keyboard gesture. This is to make the view to resign from being the first responder.